File: Algorithms/1403.Minimum-Subsequence-in-Non-Increasing-Order/minimum_subsequence_in_non_increasing_order.go
import "sort"

func minSubsequence(nums []int) []int {
	sort.Sort(sort.Reverse(sort.IntSlice(nums)))
	total := 0
	for _, ele := range nums {
		total = total + ele
	}
	var res []int
	cur := 0
	for _, ele := range nums {
		cur = cur + ele
		if cur > total-cur {
			res = append(res, ele)
			return res
		}
		res = append(res, ele)
	}
	return res
}

File: Algorithms/1403.Minimum-Subsequence-in-Non-Increasing-Order/minimum_subsequence_in_non_increasing_order.py
class Solution:
    def minSubsequence(self, nums: List[int]) -> List[int]:
        total = sum(nums)
        nums.sort(reverse=True)
        res = []
        cur = 0
        for ele in nums:
            cur += ele
            if cur > total - cur:
                res.append(ele)
                break
            res.append(ele)
        return res
